File: incompatibleCrops.swift
import Foundation

let input = readLine()!.split(separator: " ").map{Int($0)!}
let R = input[0]
let C = input[1]

var grid = [[Character]]()
for _ in 0..<R{
    let row = Array(readLine()!)
    grid.append(row)
}

let dx = [-1, 1, 0, 0]
let dy = [0, 0 , -1, 1]

var safeSpots = 0
for i in 0..<R{
    for j in 0..<C{
        if grid[i][j] == "."{
            var isSafe = true
            for k in 0..<4{
                let ni = i+dx[k]
                let nj = j+dy[k]
                if ni >= 0 && ni < R && nj >= 0 && nj < C && grid[ni][nj] == "*"{
                    isSafe = false
                    break
                }
            }
                if isSafe{
        safeSpots += 1
            }
        }
    }
}
print(safeSpots)

File: incompatibleCrops.c
#include <stdio.h>

int main(){
    int R,C;
    scanf("%d %d", &R, &C);
    char grid[R][C];
    for (int i=0; i<R; i++){
        scanf("%s", grid[i]);
    }
    int dx[] = {-1, 1, 0, 0};
    int dy[] = {0,0,-1,1};

    int safeSpots = 0;

    for (int i=0; i<R; i++){
        for(int j=0; j<C; j++){
            if(grid[i][j] == '.'){
                int isSafe = 1;

                for (int k=0; k<4; k++){
                    int newX = i+dx[k];
                    int newY = j+dy[k];

                    if(newX>=0 && newX<R && newY>=0 && newY<C && grid[newX][newY] == '*'){
                        isSafe = 0;
                        break;
                    }
                }
                if (isSafe){
                    safeSpots++;
                }
            }
        }
    }
    printf("%d\n", safeSpots);

    return 0;
}